aten/src/ATen/SavedTensorHooks.cpp

```diff
@@ -51,26 +46,26 @@ void SavedTensorDefaultHooks::set_tls_state(const impl::SavedTensorDefaultHooksT
   tls = state;
 }
 
-void SavedTensorDefaultHooks::enable() {
-  is_enabled = true;
+void SavedTensorDefaultHooks::lazy_initialize() {
```

Comment:
question: if this function is called already lazily, does it make sense for its name to still contain "lazy", because that kind of makes me think that the function itself doesn't set the flag immediately.

Reply:
The reason why I called it `lazy_initialize()` is because `push_hooks` unconditionally calls `initialize()`. It does not make sense for something to be initialized every time it gets used, so I named it "lazy_initialize" to reflect that the initialization only happens once, on the first time "lazy initialize" is called.

That being said the lazy_initialize function does very little (it just sets a flag) and is easy to read. I can change it to initialize if you want, I don't have much of an opinion on the naming here